src/util-hugepages.c

```diff
@@ -38,15 +38,18 @@ static void SystemHugepageSnapshotDump(SystemHugepageSnapshot *s);
 
 static bool SystemHugepageSupported(void)
 {
-#if !defined __CYGWIN__ && !defined OS_WIN32 && !defined __OpenBSD__ && !defined sun
+#if !defined __CYGWIN__ && !defined OS_WIN32 && !defined __OpenBSD__ && !defined sun &&            \
+        !defined __FreeBSD__
```

Comment:
Should we maybe define this by what we know works, rather than what we know doesn't work?

Reply:
yeah, I would like to put there just `#if defined __linux__` but I was afraid it can be triggered elsewhere, e.g. WSL or maybe some other OS?

Reply:
> yeah, I would like to put there just `#if defined __linux__` but I was afraid it can be triggered elsewhere, e.g. WSL or maybe some other OS?

But is that any worse than this? Being triggered on everything else we don't have covered here?

WSL is tricky, I wouldn't worry about WSL1. Not even AF_PACKET works there and we don't do anything to prevent it.  And WSL2 is true Linux with a Linux kernel, just running in a VM.